integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,910 @@
+      Integration.promotionViewed = Integration.promotionViewedEnhanced;
+      Integration.promotionClicked = Integration.promotionClickedEnhanced;
+      Integration.checkoutStarted = Integration.checkoutStartedEnhanced;
+      Integration.orderRefunded = Integration.orderRefundedEnhanced;
+      Integration.orderCompleted = Integration.orderCompletedEnhanced;
+      Integration.checkoutStepViewed = Integration.checkoutStepViewedEnhanced;
+      Integration.checkoutStepCompleted =
+        Integration.checkoutStepCompletedEnhanced;
+      Integration.orderUpdated = Integration.orderUpdatedEnhanced;
+
+      // Additional event on top og GA destination
+      Integration.productAddedToWishlist = Integration.productAddedToWishlistEnhanced;
+      Integration.productShared = Integration.productSharedEnhanced;
+      Integration.productsSearched = Integration.productsSearchedEnhanced;
+
+      // This mapping is for events which are not supported by segment and will
+      // be tracked by track method.
+      Integration.customEventsMapping = {
+        login: Integration.loggedInEnhanced,
+        sign_up: Integration.signedUpEnhanced,
+        exception: Integration.exceptionOccuredEnhanced,
+        timing_complete: Integration.timingCompletedEnhanced,
+        generate_lead: Integration.leadGeneratedEnhanced,
+        set_checkout_option: Integration.setCheckoutOptionEnhanced
+      };
+    }
+    /* eslint-enable */
+});
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/gtagjs
+ *
+ * @api public
+ */
+
+GTAG.prototype.initialize = function() {
+  var config = [];
+  var that = this;
+  var gaOptions = this.options.gaOptions;
+  var gaWebMeasurementId = this.options.gaWebMeasurementId;
+  var gaWebAppMeasurementId = this.options.gaWebAppMeasurementId;
+  var awConversionId = this.options.awConversionId;
+  var dcFloodLightId = this.options.dcFloodLightId;
+  var accountId =
+    gaWebMeasurementId ||
+    gaWebAppMeasurementId ||
+    awConversionId ||
+    dcFloodLightId;
+  if (gaWebMeasurementId || gaWebAppMeasurementId) {
+    var gaSetting = {};
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      // To Set persistent values we need to use set instead of config
+      // https://developers.google.com/analytics/devguides/collection/gtagjs/setting-values
+
+      gaSetting.custom_map = merge(
+        gaOptions.dimensions,
+        gaOptions.metrics,
+        gaOptions.contentGroupings
+      );
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/ip-anonymization
+    if (gaOptions.anonymizeIp) {
+      gaSetting.anonymize_ip = true;
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/cookies-user-id
+    if (gaOptions.domain) {
+      gaSetting.cookie_domain = gaOptions.domain;
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-link-attribution
```

Comment:
This is org analytics.js can wee update for gtag docs: https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-link-attribution